### DISCUSSION TITLE
On premises Win Server  and OpenID Connect

### DISCUSSION THREAD

**Discussion by JCV67** [Budibase/budibase#6825] [Help]
Hello,
I am trying to use OpenID Connect to allow AD users to login on my self hosted budibase. I use Caddy and the https certificate was distributed through Group Policy. I use on premises Windows Server 2022. Added ADFS server role and a budibase Server Application and Web API to it. But I still get the following message when trying to login:
{"message":"Error constructing OIDC authentication configuration","status":500}

Can you help me?
A step by step instruction would be great

Thank you
Jose

**Comment by melohagan** [comment]
Hi @JCV67 

Do either of these discussions help:
- https://github.com/Budibase/budibase/discussions/5807
- https://github.com/Budibase/budibase/discussions/6689

**Comment by JCV67** [comment]
Hi @melohagan 
I read both of them before, as well a number of OpenID and ADFS related documents on the Internet but I could not solve my problem, did not now what else to to.

**Comment by deanhannigan** [comment]
Hi @JCV67! Would it be possible to verify that the OIDC config url is reachable from the Budibase installation location?

**Reply by deanhannigan** [reply]
Hi @JCV67. Nice, seems to connect without issue and return JSON.

- Does the body of the JSON response from the curl request contain valid OIDC config?
  - Authorization requires that all of the following properties are set : `issuer, authorization_endpoint, token_endpoint, userinfo_endpoint,`
- Can you provide/review any logs recorded during an attempt to authenticate?
  - If you look for "Error constructing OIDC authentication configuration", the preceding error should indicate the root cause of the issue.

**Reply by JCV67** [reply]
Hi @deanhannigan , thank you for your help. I've been out. I can't  Budibase server OIDC logs.
